eng/native/gen-buildsys.sh

```diff
@@ -59,7 +59,7 @@ for i in "${@:6}"; do
     esac
 done
 
-cmake_extra_defines=
+cmake_extra_defines=()
```

Comment:
The key additions are on line 100, which is a translation of the WASI-specific part of gen-buildsys.cmd.  Also, incrementally accumulating arguments like we do in `cmake_extra_defines` requires some extra care in `bash` when some of those arguments have space characters in them.  Previously, we didn't need to worry about that, but the `-DCMAKE_CROSSCOMPILING_EMULATOR=...` argument _does_ have spaces, so in order to make that work I changed `cmake_extra_defines` from a string into an array.

Reply:
I looked at this a bit closer, and it looks like this script is only used in upstream build by `build-native.sh`, which does the same thing manually.

https://github.com/dotnet/runtime/blob/64b93517eb4bea5de51aa5bd7e5b32076198f1f9/src/native/libs/build-native.sh#L68-L79

Do we need `-DCMAKE_CROSSCOMPILING_EMULATOR=node ...`? I suspect we don't.

Reply:
I wasn't able to get the build to work without `-DCMAKE_CROSSCOMPILING_EMULATOR=node ...`.  I believe it's used as part of CMake platform feature detection (i.e. the little test programs that CMake builds and runs to verify that the platform supports various features).

Reply:
Interesting, in my local experiment, omitting this from the Windows build still succeeded. Let's leave investigating that for another day.

Changing the type `cmake_extra_defines` has a big diff. It would be better to add a new local (`cmake_extra_defines2`...) that has the argument with spaces in it and add that to the final cmake command line.